dhis-2/dhis-web/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/user/MeController.java

```diff
@@ -254,6 +254,37 @@ public void updateCurrentUser( HttpServletRequest request, HttpServletResponse r
         nodeService.serialize( NodeUtils.createRootNode( collectionNode.getChildren().get( 0 ) ), "application/json", response.getOutputStream() );
     }
 
+    @RequestMapping( value = "/changePassword", method = RequestMethod.PUT , consumes = { "text/*", "application/*" } )
+    @ResponseStatus( HttpStatus.ACCEPTED )
+    public void changePassword( @RequestBody Map<String, String> body, HttpServletResponse response )
+            throws WebMessageException, NotAuthenticatedException
+    {
+        User currentUser = currentUserService.getCurrentUser();
+
+        if ( currentUser == null )
+        {
+            throw new NotAuthenticatedException();
+        }
+
+        String oldPassword = body.get( "oldPassword" );
```

Comment:
Since we now refer to currentPassword in error message, maybe change this as well?